hack/prom-rule-ci/prom-rules-tests.yaml

```diff
@@ -136,3 +136,52 @@ tests:
               summary: "More than 80% of the rest calls failed in virt-handler for the last 5 minutes"
             exp_labels:
               pod: "virt-handler-1"
+
+  # Some nodes without KVM resources
+  - interval: 1m
+    input_series:
+      - series: 'kube_node_status_allocatable{resource="devices_kubevirt_io_kvm", node ="node1"}'
+        values: "110 110 110 110 110 110"
+      - series: 'kube_node_status_allocatable{resource="devices_kubevirt_io_kvm", node ="node2 "}'
+        values: "0 0 0 0 0 0"
+
+    alert_rule_test:
+      - eval_time: 5m
+        alertname: LowKVMNodesCount
+        exp_alerts:
+          - exp_annotations:
+              description: "Low number of nodes with KVM resource available."
+              summary: "At least two nodes with kvm resource required for VM life migration."
+            exp_labels:
+              severity: "warning"
+
+  # All nodes without KVM resources
+  - interval: 1m
+    input_series:
+      - series: 'kube_node_status_allocatable{resource="devices_kubevirt_io_kvm", node ="node1"}'
+        values: "0 0 0 0 0 0"
+      - series: 'kube_node_status_allocatable{resource="devices_kubevirt_io_kvm", node ="node2 "}'
+        values: "0 0 0 0 0 0"
+
+    alert_rule_test:
+      - eval_time: 5m
+        alertname: LowKVMNodesCount
+        exp_alerts:
+          - exp_annotations:
+              description: "Low number of nodes with KVM resource available."
+              summary: "At least two nodes with kvm resource required for VM life migration."
+            exp_labels:
+              severity: "warning"
+
+  # Two nodes with KVM resources
+  - interval: 1m
```

Comment:
This test should have been triggered by the `LowKVMNodesCount` alert, right? I am now wondering why it didn't happen.

Reply:
When all nodes have KVM>0 , the query  'kube_node_status_allocatable{resource=\"devices_kubevirt_io_kvm\"} == 0' will return 'no data'(and not 0 as may be expected) so the alert will not be triggered (which is fine in our case).

Reply:
Will record `num_of_kvm_available_nodes` be evaluated to `num_of_allocatable_nodes`? If so, the previous alert "(num_of_allocatable_nodes > 1) and (num_of_kvm_available_nodes < 3)" would be triggered, right?

Reply:
No.  `num_of_kvm_available_nodes` will not be evaluated to `num_of_allocatable_nodes`. It will also be 'no data'

Reply:
Thanks for the clarification!

Reply:
Thank you for good question. It is now more clear to me to.